### PR TITLE
PBI-00: reconfigure coverage testing

### PR DESCRIPTION
**Pull request by rashadaziz**
Setup coverage report for sonarcloud using jest.

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -1,2 +1,4 @@
 sonar.projectKey=one-day-intern_odi-assessee-fe
 sonar.organization=one-day-intern
+sonar.javascript.lcov.reportPaths=./coverage/lcov.info
+sonar.exclusions= **/index.ts, **/cypress/**, **/*.d.ts, **/components/shared/**/svg/**, **/*.config.js, **/*.config.ts
```
Comment (johaneschristian) [inline sonar-project.properties:4]: Please ensure that the excluded files will not contain any functionality related code.

File: .github/workflows/deploy-production.yml [hunk 2]
```diff
@@ -30,6 +30,10 @@ jobs:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Install Dependencies
+        run: npm install
+      - name: Test and coverage
+        run: npm run test:coverage
```
Comment (johaneschristian) [inline .github/workflows/deploy-production.yml:36]: If possible, reuse the test coverage information from the previous job to avoid duplication.

Comment (rashadaziz) [reply .github/workflows/deploy-production.yml:36]: image
Hello Jo, we have tried that approach however, after spending an entire day trying to figure out how to do it, we came across the sonarcloud docs and the recommended way to generate the coverage report was like this.

File: sonar-project.properties (hunk 1 above)
Comment (rashadaziz) [reply sonar-project.properties:4]: Okay Jo my bad, I accidentally added the wrong file to the sonarcloud exclusions

**Pull request merged**